game-core/src/main/java/games/strategy/engine/data/changefactory/AddUnits.java

```diff
@@ -15,19 +20,31 @@
   private final String name;
   private final Collection<Unit> units;
   private final String type;
+  /**
+   * The unit's owner can be modified sometime after this Change is created but before it is
+   * performed. To ensure that the newly created units have the correct ownership, their original
+   * owners are stored in this separate map.
+   */
+  private Map<UUID, String> unitOwnerMap;
 
   AddUnits(final UnitCollection collection, final Collection<Unit> units) {
-    this.units = new ArrayList<>(units);
+    this.units = units;
+    unitOwnerMap = buildUnitOwnerMap(units);
     name = collection.getHolder().getName();
     type = collection.getHolder().getType();
   }
 
   AddUnits(final String name, final String type, final Collection<Unit> units) {
-    this.units = new ArrayList<>(units);
+    this.units = units;
+    unitOwnerMap = buildUnitOwnerMap(units);
     this.type = type;
     this.name = name;
   }
 
+  private Map<UUID, String> buildUnitOwnerMap(final Collection<Unit> units) {
```

Comment:
side-note / big nit, this method could be static. I think we need to try and see if we can share a config that turns on the static analysis warning that methods can be marked as static.

For me it's really helpful as I watch for private methods being static. If they are, I know they do not interact with the class state. If they are not static, I assume they cannot be static and hence interact with class state.